pipeline/merger/src/main/scala/uk/ac/wellcome/platform/merger/rules/ImagesRule.scala

```diff
@@ -0,0 +1,92 @@
+package uk.ac.wellcome.platform.merger.rules
+import uk.ac.wellcome.models.work.internal.{
+  Identifiable,
+  ImageData,
+  MergedImage,
+  TransformedBaseWork,
+  UnidentifiedWork,
+  Unminted
+}
+import uk.ac.wellcome.platform.merger.models.FieldMergeResult
+import uk.ac.wellcome.platform.merger.rules.WorkPredicates.{
+  not,
+  WorkPredicate,
+  WorkPredicateOps
+}
+
+import scala.Function.const
+
+object ImagesRule extends FieldMergeRule {
+  type FieldData = List[MergedImage[Unminted]]
+
+  override def merge(
+    target: UnidentifiedWork,
+    sources: Seq[TransformedBaseWork] = Nil): FieldMergeResult[FieldData] =
+    FieldMergeResult(
+      fieldData = sources match {
+        case Nil =>
+          getSingleMiroImage.applyOrElse(target, const(Nil))
+        case _ :: _ =>
+          getPictureImages.applyOrElse((target, sources), const(Nil)) ++
+            getPairedMiroImages.applyOrElse((target, sources), const(Nil))
+      },
+      redirects = Nil
+    )
+
+  private lazy val getSingleMiroImage
+    : PartialFunction[UnidentifiedWork, FieldData] = {
+    case target if WorkPredicates.miroWork(target) =>
+      target.data.images.map {
+        _.mergeWith(
+          ImageData(
+            parentWork = Identifiable(target.sourceIdentifier),
+            fullText = createFulltext(List(target))
+          )
+        )
+      }
+  }
+
+  private lazy val getPictureImages = new FlatImageMergeRule {
+    val isDefinedForTarget: WorkPredicate = WorkPredicates.sierraPicture
+    val isDefinedForSource
+      : WorkPredicate = WorkPredicates.metsWork or WorkPredicates.miroWork
+  }
+
+  private lazy val getPairedMiroImages = new FlatImageMergeRule {
+    val isDefinedForTarget: WorkPredicate =
+      WorkPredicates.sierraWork and not(WorkPredicates.sierraPicture)
+    val isDefinedForSource: WorkPredicate = WorkPredicates.miroWork
+  }
+
+  trait FlatImageMergeRule extends PartialRule {
+    final override def rule(
+      target: UnidentifiedWork,
+      sources: Seq[TransformedBaseWork]): List[MergedImage[Unminted]] =
+      (target +: sources).flatMap {
+        _.data.images.map {
+          _.mergeWith(
+            ImageData(
+              parentWork = Identifiable(target.sourceIdentifier),
+              fullText = createFulltext(target +: sources)
+            )
+          )
+        }
+      }.toList
+  }
+
+  private def createFulltext(works: Seq[TransformedBaseWork]): Option[String] =
+    works
+      .map(_.data)
+      .flatMap { data =>
+        List(
+          data.title,
+          data.description,
+          data.physicalDescription,
+          data.lettering
```

Comment:
Would you want to put IDs of miro / sierra etc in here?

Reply:
Sorry, I know you said to not think of the shape that much...

Reply:
Quite possibly/probably - I think the contents of `Image`s will be a big ongoing discussion that will mostly be informed by search performance.

It also seems likely that `fullText` will turn out to be too limited, or at least that it shouldn't be constructed in the merger. Would prefer to leave this for now whilst being very clear that we'll come back and improve it.

Reply:
Just as a point so it doesn't get lost, having someone know why an image has come up is very important.

Searching for something that could be ambivalently neutral or derogatory is a major issue i.e. surfacing images when someone searches for animals (a case that happens). If this information isn't there to show why it was surfaced it might be assumed we are saying 👈 is true, rather than having been catalogued historically like that.

Makes me wonder if we should remove this and leave it TBD until the discussion can happen more thoroughly.

Ping @harrisonpim